acceptance/openstack/blockstorage/v3/blockstorage.go

```diff
@@ -110,6 +146,36 @@ func CreateVolumeType(t *testing.T, client *gophercloud.ServiceClient) (*volumet
 	return vt, nil
 }
 
+// CreateVolumeTypeNoExtraSpecs will create a volume type with a random name and
+// no extra specs. This is required to bypass cinder-scheduler filters and be able
+// to create a volume with this volumeType. An error will be returned if the volume
+// type was unable to be created.
+func CreateVolumeTypeNoExtraSpecs(t *testing.T, client *gophercloud.ServiceClient) (*volumetypes.VolumeType, error) {
```

Comment:
VolumeType needs to have empty extra specs in order to be able to create a volume with this type. Otherwise the cinder-scheduler and filterOpts get in the middle and the volume cannot be created. Currently on `CreateVolumeType` the extra specs are not tested due to some limitations, so it could be incorporated but i'm not sure if its the right move.

Reply:
I think the limitations have to do with the OpenLab testing environment? If so it's OK to pass on doing that kind of test. As long as functionality was tested some other way - such as a local devstack environment.

Reply:
In my understanding, if you set extra_specs then the cinder-scheduler tries to map the volumeType to a back-end that matches these extra_specs. That behavior can be different per cloud depending on how the cinder-scheduler has been configured (which [filters](https://opendev.org/openstack/cinder/src/branch/master/cinder/scheduler/filters) to use etc). I'd still say that the acceptance test is useful and i expect without extra_specs to avoid all the cloud-specific limitations of how the backens and cinder-scheduler is set-up.

I tested this on our test cluster and the acceptance passed (volume created properly, and type changed without an issue etc). I've also run a retype, basically the example on docs.go, between 2 valid volume_types that also went fine. I'll do the same with an in-use volume which should also be working